api/ocm/extensions/accessmethods/git/method.go

```diff
@@ -0,0 +1,188 @@
+
+// New creates a new git registry access spec version v1.
+func New(url, ref string, pathSpec string, opts ...AccessSpecOptions) *AccessSpec {
+	s := &AccessSpec{
+		ObjectVersionedType: runtime.NewVersionedTypedObject(Type),
+		RepoURL:             url,
+		Ref:                 ref,
+		PathSpec:            pathSpec,
+	}
+	for _, o := range opts {
+		o(s)
+	}
+	return s
+}
+
+func (a *AccessSpec) Describe(internal.Context) string {
+	return fmt.Sprintf("git commit %s[%s]", a.RepoURL, a.Ref)
+}
+
+func (*AccessSpec) IsLocal(internal.Context) bool {
+	return false
+}
+
+func (a *AccessSpec) GlobalAccessSpec(accspeccpi.Context) accspeccpi.AccessSpec {
+	return a
+}
+
+func (*AccessSpec) GetType() string {
+	return Type
+}
+
+func (a *AccessSpec) AccessMethod(c internal.ComponentVersionAccess) (internal.AccessMethod, error) {
+	return accspeccpi.AccessMethodForImplementation(newMethod(c, a))
+}
+
+func newMethod(componentVersionAccess internal.ComponentVersionAccess, accessSpec *AccessSpec) (accspeccpi.AccessMethodImpl, error) {
+	u, err := giturls.Parse(accessSpec.RepoURL)
+	if err != nil {
+		return nil, errors.ErrInvalidWrap(err, "repository repoURL", accessSpec.RepoURL)
+	}
+	if err := plumbing.ReferenceName(accessSpec.Ref).Validate(); err != nil {
+		return nil, errors.ErrInvalidWrap(err, "commit hash", accessSpec.Ref)
+	}
+
+	creds, cid, err := getCreds(accessSpec.RepoURL, componentVersionAccess.GetContext().CredentialsContext())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get credentials for repository %s: %w", accessSpec.RepoURL, err)
+	}
+
+	auth, err := techgit.AuthFromCredentials(creds)
+	if err != nil && !errors.Is(err, techgit.ErrNoValidGitCredentials) {
+		return nil, fmt.Errorf("failed to get auth method for repository %s: %w", accessSpec.RepoURL, err)
+	}
+
+	gitDownloader := git.NewDownloader(u.String(), accessSpec.Ref, accessSpec.PathSpec, auth)
+	cachedGitBlobAccessor := accessobj.CachedBlobAccessForWriter(
+		componentVersionAccess.GetContext(),
+		mime.MIME_OCTET,
+		accessio.NewWriteAtWriter(gitDownloader.Download),
+	)
+	jointCloser := func() error {
+		return errors.Join(gitDownloader.Close(), cachedGitBlobAccessor.Close())
+	}
+
+	return &accessMethod{
+		spec:   accessSpec,
+		access: cachedGitBlobAccessor,
+		close:  jointCloser,
+		cid:    cid,
+	}, nil
+}
+
+type accessMethod struct {
+	spec   *AccessSpec
+	access blobaccess.BlobAccess
+	close  func() error
+
+	cid credentials.ConsumerIdentity
+}
+
+var _ accspeccpi.AccessMethodImpl = &accessMethod{}
+
+func (m *accessMethod) Close() error {
+	if m.access == nil {
+		return nil
+	}
+
+	var err error
+	if m.close != nil {
+		err = m.close()
+	}
+	err = errors.Join(err, m.access.Close())
+
+	return err
+}
+
+func (m *accessMethod) Get() ([]byte, error) {
+	return m.access.Get()
+}
+
+func (m *accessMethod) Reader() (io.ReadCloser, error) {
+	return m.access.Reader()
+}
+
+func (m *accessMethod) MimeType() string {
+	return mime.MIME_OCTET
```

Comment:
see comment for downloader.

Reply:
This was changed to incorporate tar.gz archives instead. I still need to incorporate the path filtering